src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureSubscriptionStep.test.ts

```diff
@@ -0,0 +1,176 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { h } from 'react-hyperscript-helpers'
+import { Ajax } from 'src/libs/ajax'
+import {
+  AzureSubscriptionStep
+} from 'src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureSubscriptionStep'
+import { asMockedFn } from 'src/testing/test-utils'
+import { v4 as uuid } from 'uuid'
+
+
+type AjaxContract = ReturnType<typeof Ajax>
+jest.mock('src/libs/ajax')
+
+const getSubscriptionInput = () => screen.getByLabelText('Enter your Azure subscription ID *')
+const getManagedAppInput = () => screen.getByLabelText('Unassigned managed application *')
+
+const verifyDisabled = item => expect(item).toHaveAttribute('disabled')
+const verifyEnabled = item => expect(item).not.toHaveAttribute('disabled')
+
+// Exported for wizard integration test.
+export const selectManagedApp = async (captureEvent = jest.fn(), createAzureProject = jest.fn()) => {
+  const appName = 'appName'
+  const appRegion = 'appRegion'
+  const tenant = 'tenant'
+  const subscription = uuid()
+  const mrg = 'mrg'
+  const selectedManagedApp = { applicationDeploymentName: appName, tenantId: tenant, subscriptionId: subscription, managedResourceGroupId: mrg, assigned: false, region: appRegion }
+  const listAzureManagedApplications = jest.fn(() => Promise.resolve(
+    {
+      managedApps: [
+        { applicationDeploymentName: 'testApp1', tenantId: 'fakeTenant1', subscriptionId: 'fakeSub1', managedResourceGroupId: 'fakeMrg1', assigned: false },
+        selectedManagedApp
+      ]
+    }
+  ))
+  asMockedFn(Ajax).mockImplementation(() => ({
+    Billing: { listAzureManagedApplications, createAzureProject } as Partial<AjaxContract['Billing']>,
+    Metrics: { captureEvent } as Partial<AjaxContract['Metrics']>
+  } as Partial<AjaxContract> as AjaxContract))
+
+  // Act - Supply valid subscription UUID and wait for Ajax response
+  fireEvent.change(getSubscriptionInput(), { target: { value: subscription } })
+  await waitFor(() => verifyEnabled(getManagedAppInput()))
+
+  // Act - Select one of the managed apps
+  await userEvent.click(getManagedAppInput())
+  const selectOption = await screen.findByText(`${appName} (${appRegion})`)
+  await userEvent.click(selectOption)
+  return selectedManagedApp
+}
+
+describe('AzureSubscriptionStep', () => {
+  let onManagedAppSelectedEvent
+  let renderResult
+
+  const renderAzureSubscriptionStep = props => {
```

Comment:
I thought I was going to need to set some props separately for some tests. It turned out it wasn't necessary, but I figured I'd leave the ability anyway.

Reply:
Ha, I was just coming to the PR to comment that this method wasn't necessary and the test could be simplified by doing the setup in `beforeEach`. But I don't feel that strongly about it.